### PR TITLE
Add default service config for catalog client

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Add the capability to config catalog gRPC client with default service config.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
This is similar as https://github.com/flyteorg/flyteidl/pull/301 to add the capability to config catalog gRPC client with default service config.

Note that this is PR towards https://github.com/flyteorg/flytepropeller/pull/447

## Tracking Issue

Partially address flyteorg/flyte#2510

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/controller/nodes/task/catalog/config.go [hunk 1]
```diff
@@ -48,7 +54,7 @@ func NewCatalogClient(ctx context.Context, authOpt grpc.DialOption) (catalog.Cli
 
 	switch catalogConfig.Type {
 	case DataCatalogType:
-		return datacatalog.NewDataCatalog(ctx, catalogConfig.Endpoint, catalogConfig.Insecure, catalogConfig.MaxCacheAge.Duration, catalogConfig.UseAdminAuth, authOpt)
+		return datacatalog.NewDataCatalog(ctx, catalogConfig.Endpoint, catalogConfig.Insecure, catalogConfig.MaxCacheAge.Duration, catalogConfig.UseAdminAuth, catalogConfig.DefaultServiceConfig, authOpt)
```
Comment (honnix) [inline pkg/controller/nodes/task/catalog/config.go:57]: It is not possible to pass in the whole config directly without a refactoring because it introduces circular imports.

**Review by EngHabu — APPROVED** [review]
